velox/docs/stats.rst

```diff
@@ -0,0 +1,65 @@
+==============
+Stats Counters
+==============
+
+Memory Management
+-----------------
+.. list-table::
+   :widths: 40 10 50
+   :header-rows: 1
+
+   * - Counter Name
+     - Type
+     - Description
+   * - velox.cache_shrink_count
+     - Count
+     - The number of times that in-memory data cache has been shrunk under memory pressure.
+   * - velox.cache_shrink_ms
+     - Histogram
+     - The distribution of cache shrink latency in range of [0, 100s] with 10 buckets, and reports P50, P90, P99,
+       and P100.
+   * - velox.memory_reclaim_exec_ms
+     - Histogram
+     - The distribution of memory reclaim execution time in range of [0, 600s] with 20 buckets and reports P50, P90,
+       P99, and P100.
+   * - velox.memory_reclaim_bytes
+     - Sum
+     - The sum of reclaimed memory bytes.
+   * - velox.memory_reclaim_wait_ms
+     - Histogram
+     - The distribution of memory reclaim wait time in range of [0, 60s] with 10 buckets and reports P50, P90, P99,
+       and P100.
+   * - velox.memory_reclaim_wait_timeout_count
+     - Count
+     - The number of times that the memory reclaim wait timeouts.
+   * - velox.memory_non_reclaimable_count
+     - Count
+     - The number of times that the memory reclaim fails because of non-reclaimable section which is an indicator that
```

Comment:
"The number of times that the memory reclaim fails because of the non-reclaimable section, which indicates that the memory reservation is insufficient." ?